src/rpc/blockchain.cpp

```diff
@@ -2234,6 +2235,40 @@ class BlockFiltersScanReserver
     }
 };
 
+static bool CheckBlockFilterMatches(const CBlockIndex* pindex, const GCSFilter::ElementSet& needles, const CChainParams& chainparams)
+{
+    CBlock block;
+    if (!ReadBlockFromDisk(block, pindex, chainparams.GetConsensus())) {
+        throw JSONRPCError(RPC_DATABASE_ERROR, "Cannot read block from disk");
+    }
+
+    CBlockUndo block_undo;
+    // Genesis block has no undo data
+    if (pindex->nHeight > 0 && !UndoReadFromDisk(block_undo, pindex)) {
+        throw JSONRPCError(RPC_DATABASE_ERROR, "Cannot read block undo data from disk");
+    }
```

Comment:
Would be nice to use the helper functions `GetBlockChecked` and `GetUndoChecked` here to have consistent error messages and RPC return codes and avoid repetition. The only drawback right now is that `GetUndoChecked` doesn't include the exception for the genesis blocks yet, so the test would fail. Fixed by #19888 or #26369.

Reply:
Pushed suggestion https://github.com/bitcoin/bitcoin/compare/33793d67bff6519b6ebdd0b1229d03166c505672..50a145c39986dd7ed4a5667d6433c9f590b29763